packages/blade/src/components/Typography/Text/Text.tsx

```diff
@@ -0,0 +1,101 @@
+import type { ReactElement } from 'react';
+import styled from 'styled-components';
+import type { TextTypes } from '../../../tokens/theme/theme';
+import getPlatformType from '../../../utils/getPlatformType';
+import type { Theme } from '../../BladeProvider';
+import BaseText from '../BaseText';
+import type { BaseTextProps } from '../BaseText/BaseText';
+
+type TextCommonProps = {
+  type: TextTypes;
+  truncateAfterLines?: number;
+  children?: React.ReactNode;
+};
+
+type TextBodyVariant = TextCommonProps & {
+  variant: 'body';
+  weight: keyof Theme['typography']['fonts']['weight'];
+};
+
+type TextCaptionVariant = TextCommonProps & {
+  variant: 'caption';
+  weight: keyof Pick<Theme['typography']['fonts']['weight'], 'regular'>;
+};
+
+export type TextProps<T> = T extends {
```

Comment:
We should probably add a small description here, Why we are doing it this way instead of overloads or unions. 

```ts
/**
 * Conditionally changing props based on variant
 * Because overloads or union gives wrong intellisense.
 */
```

Reply:
aye sir

Reply:
done